### ISSUE TITLE
[Latex] syntax highlighting breaks with \newcommand with \begin

### ISSUE THREAD

**Issue by neilernst** [sublimehq/Packages#518]
This is hopefully fixed by the rewrite (#340), but another example that fails currently is 
```
\newcommand{\bi}{\begin{itemize}}
```

as the end of the `\newcommand` command is not detected, so we remain in the `meta.group.braces.tex` and  `meta.function.environment.list.latex` scopes.

This also occurs with 
```
\newcommand{\bg}{\begin{equation}}
```
mentioned by @randy3k in #340

**Comment by randy3k** [comment]
They are fixed in #370

**Comment by wbond** [comment]
Closing, as this is addressed in #370.
